packages/builder/src/builtin-tasks/compile.ts

```diff
@@ -0,0 +1,72 @@
+import { task } from "../internal/core/config/config-env";
+import { networks } from "../../sample-project/algob.config";
+import { TASK_COMPILE } from "./task-names";
+import YAML from "yaml";
+const algosdk = require('algosdk');
+const fs = require('fs');
+const path = require('path');
+const murmurhash = require('murmurhash');
+
+/* Credentials */
+const token = networks.localhost.token;
+const host = networks.localhost.host;
+const port = networks.localhost.port;
+
+let algodclient = new algosdk.Algodv2(token, host, port);
+
+export default function (): void {
+  task(TASK_COMPILE, "Compilation task")
+    .setAction(
+      async () => {
+        console.log("compiling......");
+        /*File paths for assets/ and artifacts/cache/ */
+        var srcPath = path.join(__dirname, '../../sample-project/assets/');
+        var destPath = path.join(__dirname, '../../sample-project/artifacts/cache');
+        /*Read files from assets/ and artifacts/cache/ */
+        const assetsFiles = fs.readdirSync(srcPath);
+        const compiledFiles = fs.readdirSync(destPath);
+        const compiledSrcHashes = [];
+        /* Store all source file hash from already compiled files in an array */
+    	for(let cfile in compiledFiles) {
+    		let pathCFile = path.join(destPath, compiledFiles[cfile]);
+    		let cdata = YAML.parse(fs.readFileSync(pathCFile).toString());
+    		console.log("compiled file data", cdata.srcHash);
+    		compiledSrcHashes.push(cdata.srcHash);
```

Comment:
You rely on array indices. It works but it's not readable/understandable.
Can you have a method that returns an object of:
```
{
  hash: "murmurhash",
  compiledFile: "data",
  maybeAssetFile: "data" | undefined
  ...
}
```
This will reduce the amount of `for` loops and will be more readable -- also it could be refactored to a lazy version later on.